src/Sylius/Bundle/PromotionBundle/Form/Type/PromotionCouponGeneratorInstructionType.php

```diff
@@ -13,14 +13,26 @@
 
 namespace Sylius\Bundle\PromotionBundle\Form\Type;
 
-use Sylius\Bundle\ResourceBundle\Form\Type\AbstractResourceType;
+use Sylius\Component\Promotion\Factory\PromotionCouponGeneratorInstructionFactoryInterface;
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\DataMapperInterface;
 use Symfony\Component\Form\Extension\Core\Type\DateType;
 use Symfony\Component\Form\Extension\Core\Type\IntegerType;
 use Symfony\Component\Form\Extension\Core\Type\TextType;
 use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\OptionsResolver\OptionsResolver;
 
-final class PromotionCouponGeneratorInstructionType extends AbstractResourceType
+final class PromotionCouponGeneratorInstructionType extends AbstractType implements DataMapperInterface
```

Comment:
Why did you need to change from the `AbstractResourceType` to the `AbstractType` 🤔?